### ISSUE TITLE
Update/Migrate Entity Type From One Type to Another

### ISSUE THREAD

**Issue by lreading** [OWASP/threat-dragon#199]
**Describe what problem your feature request solves**
Entities are unable to be changed after they are added to a diagram (Store, Actor, Process).  This is problematic when a diagram changes or is first being defined.  Additionally, when an entity is removed from the diagram, all attached flows are removed as well.

**Describe the solution you'd like**
In the context/edit menu for an entity, there should be an option to change the entity type.  The only supported changes would be between Stores, Actors and Processes.

**Additional context**
This was suggested by Simon S

**Comment by jgadsden** [comment]
This is an excellent suggestion by Simon. A point to note is that the data flow connections will have to be preserved, and this change will have to be intuitive but not done accidentally ... a difficult thing to get right without too much clutter on the right hand pane?

**Comment by lreading** [comment]
I'm almost at the point where this could be in the near-term for v2.  In v2, data flow connection points are (mostly) managed by the underlying drawing package, and will move based on anchors and other context.  You _can_ put flows at arbitrary points on the graph, but they do snap to other objects.  When they snap to an object, instead of having absolute positions (eg x: 23, y: 250), they just reference the other object.

In short, I think it will probably work out fine in most situations, as the drawing package should be intelligent enough to make small adjustments on the fly.

**Comment by jgadsden** [comment]
It is unlikely we will be able to get this into version 2.0, so I have marked it for the later 2.x version

**Comment by jgadsden** [comment]
Closing this as a Won't Do
It would complicate the tool, Threat Dragon is deliberately lightweight compared to Microsoft TMT, and we need tokeep it simple - otherwise we start competing with MS TMT
